### PR TITLE
Improve and update how to Documentation to latest code changes.

### LINKED ISSUE #180: Improve How-to documentation
Documentation describes not clear enough the different steps.
Heading are confusing.

### PR DESCRIPTION
**Pull request by rdebusscher**
Fixes #180

### CODE REVIEW

File: how-to.md [hunk 1]
```diff
@@ -3,56 +3,82 @@
...
-* Define a Maven property in the **adaptMavenModel()** of the specific server add-on and make sure this property is used in the 'profile' to the file `src/main/resources/pom)servers.xml`.  
+* Adapt the method **adaptMavenModel()** of the specific server add-on.  
 * Test out the generated project with this new version.
    Start the Starter application (see **/Container/README.md in this repository** on how to start starter locally), select MP version and your new implementation and click on download. Test out the Maven project in the downloaded ZIP file.
 * In case the implementation requires adjusted example files (which is, however, an indication of some problem following the specification) you can create a customized template within the directory `src/main/resources/files/<code>`
-   This adjusted template file will be picked up automatically.
+   This adjusted template file will be picked up automatically when the rules described in the section **Add a new template file** are followed.
 
-## A verify specific 'profile' for a server - MicroProfile version combination
+## Use a verify specific 'profile' for an Implementation - MicroProfile version combination
```
Comment (Emily-Jiang) [inline how-to.md]: verify => verified

File: how-to.md [hunk 2]
```diff
@@ -3,56 +3,82 @@
...
 
-* The selection of the 'profile' within the `src/main/resources/pom)servers.xml` file is:
-   1. based on server and MP version -> `<code>-<version>`
-   2. If not found,  based on the server -> `<code>`
+* The selection of the 'profile' within the `src/main/resources/pom-servers.xml` file is:
+   1. based on implementation and MP version -> `<code>-<version>`
```
Comment (Emily-Jiang) [inline how-to.md]: implementation => the implementation

File: how-to.md [hunk 3]
```diff
@@ -3,56 +3,82 @@
...
 * Add an `if-structure` to the method **createFiles()** of **org.eclipse.microprofile.starter.addon.microprofile.servers.MicroprofileServersAddon** so that files are included in the demo application when the specification is selected.
 * Update the **readme.md.tpl** to include some basic information about this new specification. Make this addition conditional (see other specification on how to do this)
 
-## Add new MicroProfile Version
+## Adding a new MicroProfile Version
 
 * Add the version to the enum **org.eclipse.microprofile.starter.core.model.MicroProfileVersion**. The third parameter is only required when the maven MicroProfile version is not the same as _version value_ (as for example with version 2.0)
-* Add to each MicroProfile specification this new version number in the class **org.eclipse.microprofile.starter.addon.microprofile.servers.model.MicroprofileSpec**
+* Add to each MicroProfile specification this new version number in the class **org.eclipse.microprofile.starter.addon.microprofile.servers.model.MicroprofileSpec**. (since existing specifications will also be part of the new MicroProfile version)
```
Comment (Emily-Jiang) [inline how-to.md]: this new version => with this new version

**Comment by rdebusscher** [comment]
@Emily-Jiang Thx for reviewing.  I have addressed the comments (and did some rewrites to make things more clear)

**Comment by Karm** [comment]
Has 2 acks. Merged.

**Pull request merged**